File: GridView/PermanentDropDownListEditorInFilterCell/PermanentDropDownListEditorInFilterCellVB/CustomColumn.vb
Imports Telerik.WinControls.UI

Public Class CustomColumn
    Inherits GridViewTextBoxColumn
    Public Overrides Function GetCellType(ByVal row As GridViewRowInfo) As Type
        If TypeOf row Is GridViewFilteringRowInfo Then
            Return GetType(DropDownGridFilterCellElement)
        End If
        Return MyBase.GetCellType(row)
    End Function
End Class

File: gridview/PermanentDropDownListEditorInFilterCell/PermanentDropDownListEditorInFilterCell/CustomColumn.cs
using System;
using System.Collections.Generic;
using System.Text;
using Telerik.WinControls.UI;

namespace PermanentDropDownListEditorInFilterCell
{
    class CustomColumn : GridViewTextBoxColumn
    {
        public override Type GetCellType(GridViewRowInfo row)
        {
            if (row is GridViewFilteringRowInfo)
            {
                return typeof(DropDownGridFilterCellElement);
            }
            return base.GetCellType(row);
        }
    }
}